ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -101,4 +205,468 @@ public Integer getValue() {
     };
     metricRegistry.register(MetricRegistry.name(NonBlockingRouter.class, "-NumActiveOperations"), activeOperations);
   }
-}
+
+  /**
+   * Update opration latency for a given {@link com.github.ambry.router.NonBlockingRouter.RouterOperationType}.
+   * <p/>
+   * This method should be called when an {@code Operation} is fully completed (either successfully or unsuccessfully),
+   * which means the {@code operation} has gone through the complete process without being aborted in half way.
+   * @param latency The operation latency to be updated.
+   * @param operationType The {@link com.github.ambry.router.NonBlockingRouter.RouterOperationType}
+   *                      for which metrics will be updated.
+   */
+  void updateOperationLatency(long latency, NonBlockingRouter.RouterOperationType operationType) {
```

Comment:
this method is completely unnecessary. I absolutely don't see why you think

`updateOperationLatency(latency, NonBlockingRouter.RouterOperationType.DeleteBlob)`

is better than

`routerMetrics.deleteBlobLatency.update(latency)`